compiler/rustc_codegen_llvm/src/common.rs

```diff
@@ -316,15 +318,38 @@ impl<'ll, 'tcx> ConstCodegenMethods for CodegenCx<'ll, 'tcx> {
                                 }),
                             )))
                             .unwrap_memory();
-                        let init = const_alloc_to_llvm(self, alloc, /*static*/ false);
-                        let value = self.static_addr_of_impl(init, alloc.inner().align, None);
-                        value
+                        let init = const_alloc_to_llvm(self, alloc.inner(), /*static*/ false);
+                        self.static_addr_of_impl(init, alloc.inner().align, None)
                     }
                     GlobalAlloc::Static(def_id) => {
                         assert!(self.tcx.is_static(def_id));
                         assert!(!self.tcx.is_thread_local_static(def_id));
                         self.get_static(def_id)
                     }
+                    GlobalAlloc::Type { ty, segment } => {
```

Comment:
This code path ignores the pointer offset. Should probably error if it is not `0` instead of silently allowing people to offset it as they want without that actually doing anything

Reply:
or we add the offset to the hash value, which is technically what was done by the user